nengo/connection.py

```diff
@@ -168,18 +169,28 @@ def check_function_can_be_applied(self, conn, function_info):
         size_mid = conn.size_in if size is None else size
         transform = conn.transform
 
-        if isinstance(transform, np.ndarray):
-            if transform.ndim < 2 and size_mid != conn.size_out:
-                raise ValidationError(
-                    "function output size is incorrect; should return a "
-                    "vector of size %d" % conn.size_out, attr=self.name,
-                    obj=conn)
+        if isinstance(transform, (np.ndarray, Transform)):
```

Comment:
I think having full weights also be a transform (as discussed at the meeting) will make the following sections a lot cleaner. I'm going to hold off looking at them too closely until that's done.